kubernetes-tests/src/test/java/io/fabric8/kubernetes/client/mock/LoadTest.java

```diff
@@ -0,0 +1,69 @@
+package io.fabric8.kubernetes.client.mock;
+
+import io.fabric8.kubernetes.api.model.HasMetadata;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.server.mock.KubernetesServer;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class LoadTest {
+
+  @Rule
+  public KubernetesServer server = new KubernetesServer();
+
+  @Test
+  public void testResourceGetFromLoadWhenMultipleDocumentsWithDelimiter() throws Exception {
+    // given
+    KubernetesClient client = server.getClient();
+
+    // when
+    List<HasMetadata> result = client.load(getClass().getResourceAsStream("/multiple-document-template.yml")).get();
+
+    // then
+    assertNotNull(result);
+    assertEquals(6, result.size());
+    HasMetadata deploymentResource = result.get(1);
```

Comment:
Are we sure about the order?

Reply:
@hrishin, not sure about it, but so far I see the order being preserved.

Reply:
@hemanik thanks!